### DISCUSSION TITLE
Best way to track particles

### DISCUSSION THREAD

**Discussion by ghost** [BLAST-WarpX/warpx#4430] [Simulation Inputs]
Hi,

I'm trying to figure out the best way to track specific particles through a simulation. I'd also need information about each of the particles such as velocity, temperature, and momentum. 

Thanks!

**Comment by ax3l** [comment]
Hi @Nsh002,

there are multiple ways to track particles. Each particle has a unique id, so you can identify it later
- use a full diagnostics, and write them out or
- use the Python wrappers

Can you tell a bit more about your use case? How often and how many particles do you need?
